src/backend/CommandBuffer.cpp

```diff
@@ -70,6 +70,16 @@ namespace backend {
             return true;
         }
 
+        bool ValidateBufferOffset(CommandBufferBuilder* builder, TextureBase* texture, const BufferCopyLocation& location) {
+            uint32_t texelSize = static_cast<uint32_t>(TextureFormatPixelSize(texture->GetFormat()));
```

Comment:
I wonder how this is going to work for RGB8Unorm, probably texel size will still be 4 in that case?